src/internal/mod.rs

```diff
@@ -638,32 +649,41 @@ pub(crate) mod test {
     #[test]
     fn serde_devicecontext_roundtrip() {
         use serde_json;
-        let recrypt = recrypt::api::Recrypt::new();
-        let priv_key = recrypt.random_private_key();
+        let priv_key: recrypt::api::PrivateKey = recrypt::api::PrivateKey::new_from_slice(
+            base64::decode("bzb0Rlg0u7gx9wDuk1ppRI77OH/0ferXleenJ3Ag6Jg=")
+                .unwrap()
+                .as_slice(),
+        )
+        .unwrap();
+
         let dev_keys = recrypt::api::SigningKeypair::from_byte_slice(&[
             1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
             1, 1, 1, 138, 136, 227, 221, 116, 9, 241, 149, 253, 82, 219, 45, 60, 186, 93, 114, 202,
             103, 9, 191, 29, 148, 18, 27, 243, 116, 136, 1, 180, 15, 111, 92,
         ])
         .unwrap();
         let context = DeviceContext::new(
+            TryFrom::try_from(314).unwrap(),
             TryFrom::try_from("account_id").unwrap(),
             22,
             priv_key.into(),
             DeviceSigningKeyPair::from(dev_keys),
         );
 
-        let json = serde_json::to_string(&context);
+        let json = serde_json::to_string(&context).unwrap();
+
+        assert_eq!(json, "{\"deviceId\":314,\"accountId\":\"account_id\",\"segmentId\":22,\"signingPrivateKey\":\"AQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQGKiOPddAnxlf1S2y08ul1yymcJvx2UEhvzdIgBtA9vXA==\",\"devicePrivateKey\":\"bzb0Rlg0u7gx9wDuk1ppRI77OH/0ferXleenJ3Ag6Jg=\"}");
```

Comment:
I think you can use a "raw string" here to avoid escaping so much. 
`r#"..."`

Reply:
Much nicer. Thank you!